benches/flat_map.rs

```diff
@@ -0,0 +1,104 @@
+#![feature(test)]
+#![feature(i128_type)]
+
+extern crate rand;
+extern crate test;
+extern crate flat_map;
+
+use std::collections::BTreeMap;
+use std::iter::FromIterator;
+use flat_map::FlatMap;
+use test::Bencher;
+use rand::Rng;
+use rand::OsRng;
+
+
+type Key = u64;
+type Value = u64;
+
+const NUM_KEYS: usize = 2 << 12;
+const NUM_VALUES: usize = 1 << 12;
+
+fn flat_map_setup() -> (
+    FlatMap<Key, Value>,
+    Vec<Key>,
+) {
+    let rng = OsRng::new().unwrap();
+    let num_items = 1 << 25;
+
+    let mut keys = Vec::new();
+    let mut key_vals = Vec::with_capacity(num_items);
+    for i in 0..NUM_KEYS*NUM_VALUES {
+        // let key = rand::random::<Key>();
+        keys.push(i as u64);
+        let value = rand::random::<u64>();
+        key_vals.push((i as u64, value))
+    }
+
+    let map: FlatMap<_, _> = key_vals.into_iter().collect();
+    keys.sort();
+    keys.dedup();
+
+    (map, keys)
+}
+
+fn btree_map_setup() -> (
+    BTreeMap<Key, Value>,
+    Vec<Key>,
+) {
+    let rng = OsRng::new().unwrap();
+    let num_items = 1 << 25;
+
+    let mut keys = Vec::new();
+    let mut key_vals = Vec::with_capacity(num_items);
+    for i in 0..NUM_KEYS*NUM_VALUES {
+        // let key = rand::random::<Key>();
+        keys.push(i as u64);
+        let value = rand::random::<u64>();
+        key_vals.push((i as u64, value))
+    }
+
+    // key_vals.sort_by(|pair1, pair2|  pair1.0.cmp(&pair2.0));
+    // key_vals.dedup();
```

Comment:
I'm not a fan of committing commented out code. Can you please remove this and the two `// let key = rand::random::<Key>();` above?